### PR TITLE
OHRM5X-1160: Attendance Upgrader added

### PR DESCRIPTION
**Pull request by VageeshOrangeHRM**
This PR Contains following changes:
1. OHRM5X-1160: Attendance Upgrader added.
2. Migration file and related Doctrine Database Abstraction Layer added.

### CODE REVIEW

File: installer/upgrader/Migrations/V5/AttendanceHelper.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+     * @param string|null $punchOutTimezone
+     * @return void
+     */
+    public function updateAttendanceRecords(int $id, string $punchTimezone, ?string $punchOutTimezone = null): void
+    {
+        $q = $this->createQueryBuilder();
+        $q->update('ohrm_attendance_record')
+            ->set('ohrm_attendance_record.punch_in_timezone', ':punchInTimezone')
+            ->set('ohrm_attendance_record.punch_out_timezone', ':punchOutTimezone')
+            ->where('ohrm_attendance_record.id = :id')
```
Comment (RajithaKumara) [inline installer/upgrader/Migrations/V5/AttendanceHelper.php:99]: can't we based on 
```sql
punch_in_time_offset = :punchInTimezoneOffset AND `punch_out_time_offset` = :punchOutTimezoneOffset ;
```

File: installer/upgrader/Migrations/Version20220301.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    public function up(): void
+    {
+        //getting all attendance records from data source
+        $attendanceRecordList = $this->getAttendanceHelper()->getAttendanceRecords();
+        foreach ($attendanceRecordList as $attendanceRecords) {
+            $id = $attendanceRecords['id'];
+            $punchOutTimeZone = null;
+            $punchInTimeZone = array_search($attendanceRecords['punch_in_time_offset'], AttendanceHelper::TIMEZONE_MAP);
```
Comment (RajithaKumara) [inline installer/upgrader/Migrations/Version20220301.php:46]:
```suggestion
        $attendanceRecordList = $this->getAttendanceHelper()->getAttendanceRecords();
        $timezoneMap = array_flip(AttendanceHelper::TIMEZONE_MAP);
        foreach ($attendanceRecordList as $attendanceRecords) {
            $id = $attendanceRecords['id'];
            $punchOutTimeZone = null;
            $punchInTimeZone = $timezoneMap[$attendanceRecords['punch_in_time_offset']];
```

File: installer/upgrader/Migrations/Version20220301.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            if (!is_null($attendanceRecords['punch_out_time_offset'])) {
+                $punchOutTimeZone = array_search(
+                    $attendanceRecords['punch_out_time_offset'],
+                    AttendanceHelper::TIMEZONE_MAP
+                );
```
Comment (RajithaKumara) [inline installer/upgrader/Migrations/Version20220301.php:51]:
```suggestion
                $punchOutTimeZone = $timezoneMap[$attendanceRecords['punch_out_time_offset']];
```

File: installer/upgrader/console.php [hunk 4]
```diff
@@ -21,11 +21,13 @@
...
 use OrangeHRM\ORM\Doctrine;
 use OrangeHRM\Tools\Migrations\Version20220125;
+use OrangeHRM\Tools\Migrations\Version20220301;
 
 require_once realpath(__DIR__ . '/../../symfony/vendor/autoload.php');
 
 ServiceContainer::getContainer()->register(Services::DOCTRINE)
     ->setFactory([Doctrine::class, 'getEntityManager']);
 
 $migration = new Version20220125();
```
Comment (RajithaKumara) [inline installer/upgrader/console.php:31]:
```suggestion
$migration = new Version20220125();
$migration->up();
```

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: installer/upgrader/Migrations/Version20220301.php (hunk 2 above)
Comment (VageeshOrangeHRM) [reply installer/upgrader/Migrations/Version20220301.php:46]: i did like this, but i got warning in console. thats y i used to array_search

File: installer/upgrader/Migrations/Version20220301.php (hunk 3 above)
Comment (VageeshOrangeHRM) [reply installer/upgrader/Migrations/Version20220301.php:51]: Here also

File: installer/upgrader/console.php (hunk 4 above)
Comment (VageeshOrangeHRM) [reply installer/upgrader/console.php:31]: ???

File: installer/upgrader/Migrations/V5/AttendanceHelper.php (hunk 1 above)
Comment (VageeshOrangeHRM) [reply installer/upgrader/Migrations/V5/AttendanceHelper.php]: done

File: installer/upgrader/Migrations/Version20220301.php (hunk 2 above)
Comment (VageeshOrangeHRM) [reply installer/upgrader/Migrations/Version20220301.php]: done

File: installer/upgrader/Migrations/Version20220301.php (hunk 3 above)
Comment (VageeshOrangeHRM) [reply installer/upgrader/Migrations/Version20220301.php]: done

File: installer/upgrader/console.php (hunk 4 above)
Comment (VageeshOrangeHRM) [reply installer/upgrader/console.php:31]: done

**Review by RajithaKumara — APPROVED** [review]

**Comment by RajithaKumara** [comment]
How to test:

1. Install 4.x system and perform punch in/out operation under different timezones
2. Dump `ohrm_attendance_record` table & `hs_hr_employee`
3. Import those table data to freshly installed 5.x app
4. And run below command inside PHP container;
```bash
php installer/upgrader/console.php
```

**Pull request merged**